### PR TITLE
Searchfeeder replacement: Changes all over

### PR DESCRIPTION
**Pull request by tosan88**
Changes include:

* Model coming from PostPublicationCombiner queue updated
* Introducing call to public-concordance-api to get TME ids
* Healtcheck for public-concordance-api
* Mapping annotation-related fields changed
* Mapping content-related fields updated
* Handling deletes changed
* Processing kafka messages sequentially
* Massive refactorings
* Tests added/updated

### CODE REVIEW

File: concept/concordance.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+type concordance struct {
+	Concept    concept    `json:"concept"`
+	Identifier identifier `json:"identifier"`
+}
+
+type ConcordancesResponse struct {
+	Concordances []concordance `json:"concordances"`
+}
+
+type Model struct {
```
Comment (BerniVarga) [inline concept/concordance.go]: Not a very suggestive name, could you change it to something more suggestive?

File: concept/concordance.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+	var concordancesResp ConcordancesResponse
+	if err = json.Unmarshal(body, &concordancesResp); err != nil {
+		return nil, err
+	}
+
+	return TransformToConceptModel(concordancesResp), nil
+}
+
+func TransformToConceptModel(concordancesResp ConcordancesResponse) map[string]*Model {
```
Comment (BerniVarga) [inline concept/concordance.go]: Do we need to have the address of the Model here?

File: concept/concordance.go (hunk 1 above)
Comment (tosan88) [reply concept/concordance.go]: Yes, my bad, changing it

File: concept/concordance.go (hunk 2 above)
Comment (tosan88) [reply concept/concordance.go]: Probably not, changing it

File: concept/concordance.go [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+	return TransformToConceptModel(concordancesResp), nil
+}
+
+func TransformToConceptModel(concordancesResp ConcordancesResponse) map[string]ConceptModel {
+	conceptMap := make(map[string]ConceptModel)
+	for _, c := range concordancesResp.Concordances {
+		concept, found := conceptMap[c.Concept.ID]
+		if !found {
+			conceptMap[c.Concept.ID] = ConceptModel{}
+			concept = conceptMap[c.Concept.ID]
```
Comment (BerniVarga) [inline concept/concordance.go]: I think I would simplify here, by getting rid of the concept variable.

File: helm/content-rw-elasticsearch/values.yaml [hunk 4]
```diff
@@ -17,3 +17,4 @@ resources:
 env:
```
Comment (BerniVarga) [inline helm/content-rw-elasticsearch/values.yaml]: Are these empty values necessary?

File: messaging/mapper.go [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+			model.CmrGenreIds = prepareElasticField(model.CmrGenreIds, annIDs)
+		}
+	}
+}
+
+func (handler *MessageHandler) prepareAnnotationsWithConcepts(enrichedContent *content.EnrichedContent, tid string) ([]content.Thing, map[string]concept.ConceptModel, error) {
+	var ids []string
+	var anns []content.Thing
+	for _, a := range enrichedContent.Metadata {
+		switch a.Thing.Predicate {
```
Comment (BerniVarga) [inline messaging/mapper.go]: Couldn't we use a simple if for filtering certain predicates out?

File: concept/concordance.go (hunk 3 above)
Comment (tosan88) [reply concept/concordance.go]: We can't get rid of the variable but the code could be further simplified

File: helm/content-rw-elasticsearch/values.yaml (hunk 4 above)
Comment (tosan88) [reply helm/content-rw-elasticsearch/values.yaml]: This is the convention in our repos, so those values are enforced to be populated in the actual app files (e.g. https://github.com/Financial-Times/native-ingester/blob/master/helm/native-ingester/values.yaml#L12)

File: messaging/mapper.go (hunk 5 above)
Comment (tosan88) [reply messaging/mapper.go]: Yes, I will change that

**Comment by tosan88** [comment]
@BerniVarga I tried different configurations to organize these package before, but I'm still not satisfied with the result. I changed now based on your suggestions, hopefully it makes more sense.

**Pull request merged**